File: July-LeetCoding-Challenge/19-Add-Binary/Add-Binary.cs
public class Solution {
    public string AddBinary(string a, string b)
    {
        var sb = new StringBuilder();
        int i = a.Length-1, j = b.Length-1, carry = 0;
        while(i >=0 || j >= 0){
            int sum = carry;
            if(i >= 0) sum += a[i--] - '0';
            if(j >= 0) sum += b[j--] - '0';
            sb.Append(sum % 2);
            carry = sum /2;
        }
        if(carry != 0) sb.Append(carry);
        var arr = sb.ToString().ToCharArray();
        Array.Reverse(arr);
        return new String(arr);
    }
}

File: July-LeetCoding-Challenge/19-Add-Binary/Add-Binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string ans = (a.length() > b.length())?a:b;
        int sum = 0;
        for (int i = 0; i < ans.length(); i++){
            if (i < a.length()) sum += (a[a.length() - i - 1] == '1')?1:0;
            if (i < b.length()) sum += (b[b.length() - i - 1] == '1')?1:0;
            ans[ans.length() - i - 1] = (sum & 1) + '0';
            sum >>= 1;
        }
        if (sum == 1) ans = "1" + ans;
        return ans;
    }
};